docs/pages/api.rst

```diff
@@ -1512,10 +1561,10 @@ Signature: ``Collection::scale(float $lowerBound, float $upperBound, ?float $wan
 .. code-block:: php
 
     $collection = Collection::range(0, 10, 2)
-        ->scale(0, 10);
+        ->scale(0, 10); // [0, 0.2, 0.4, 0.6, 0.8]
```

Comment:
`scale` is another one that I'd like to improve the description of, as it's quite hard to understand what exactly it does and where it could be used. I still don't fully get it 😅

Reply:
This is a very useful function.

Let's say that you have these numbers: `[0, 5,10,15,20]`

You would like to "scale" them using a another linear scale, let's say, from 0 to 12.

```php
$collection = Collection::fromIterable([0, 5, 10, 15, 20])
  ->scale(0, 20, 0, 12); // [0, 3, 6, 9, 12]
```

You can also get a logarithmic scale by adding an extra `base` parameter

```php
$collection = Collection::fromIterable([0, 5, 10, 15, 20])
  ->scale(0, 20, 0, 12, \M_E); // [1, 6.909, 9.454, 10.943, 12]
```

Let me know if you need more explanation.

Reply:
gotcha, it has to do with [normalization](https://en.wikipedia.org/wiki/Normalization_(statistics)) and basically allows you to keep the same relative distance between values, but have different lower and upper bounds for the set of values

Reply:
That's indeed exactly it.